Fix the branch 2-4-stable has errors and failure in spec

wktk opened the pull request.

wktk commented:
I will merge the #958 commits into #949 if this PR is accepted, then the check of #949 should pass.

jeremy commented on Gemfile:
```diff
@@ -1,10 +1,10 @@
 gem "mime-types", "~> 1.16"
 gem "treetop", "~> 1.4.10"
-gem "i18n", ">= 0.4.0"
+gem "i18n", "~> 0.6.11"
```
`'>= 0.4.0', '< 0.7.0'`

jeremy commented on Gemfile:
```diff
@@ -1,10 +1,10 @@
 source :rubygems
 
-gem "activesupport", ">= 2.3.6"
+gem "activesupport", "~> 3.2", ">= 2.3.6"
```
`'>= 2.3.6', '< 4.0.0'`

jeremy commented on mail.gemspec:
```diff
@@ -17,7 +17,7 @@ Gem::Specification.new do |s|
   s.add_dependency('mime-types', "~> 1.16")
   s.add_dependency('treetop', '~> 1.4.8')
-  s.add_dependency('i18n', '>= 0.4.0')
+  s.add_dependency('i18n', '~> 0.6.11')
```
'>= 0.4.0', '< 0.7.0'

wktk commented:
Changes in `Gemfile` and `mail.gemspec` is not a backport.
Changes in `lib/mail/version_specific/ruby_1_9.rb` is a backport of bb25038, but cherry-picking bb25038 broke test when I tried, so I backported minimum code.

jeremy commented on lib/mail/version_specific/ruby_1_9.rb:
```diff
@@ -68,7 +68,7 @@ def Ruby19.q_value_decode(str)
       if match
         encoding = match[1]
-        str = Encodings::QuotedPrintable.decode(match[2].gsub(/_/, '=20'))
+        str = Encodings::QuotedPrintable.decode(match[2].gsub(/_/, '=20').gsub(/\=$/, ''))
```
What is this backporting?

Can you cherry-pick the backported commit directly, preserving the commit message, author, spec, etc?

jeremy commented on Gemfile:
```diff
@@ -1,10 +1,10 @@
 gem "mime-types", "~> 1.16"
 gem "treetop", "~> 1.4.10"
-gem "i18n", ">= 0.4.0"
+gem "i18n", '>= 0.4.0', '< 0.7.0'
```
I understand we're limiting these since newer versions require Ruby >= 1.9.3.

However, Ruby 1.9.3+ users may use the mail lib also. Should we restrict their maximum version? It would be a very incompatible change for a stable 2.4.x release.

wktk commented:
@jeremy I updated `Gemfile` and `gemspec` not to break compatibility for 1.9.3+, and cherry-picked the backported commit to fix spec.  The test of 6e12d7f has failed because `Mail::Message`s which should be equal are generated 1 second apart and regarded different, so I also cherry-picked db7c955 to solve that.  db7c955 is written in newer syntax of RSpec, so I fixed it in 0bf9f90.
Can you review?

jeremy commented:
Closed by 2a9061c
